src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -105,30 +110,49 @@ public void update(String passphrase, CAType caType) throws KeyStoreException {
             createAndStoreDefaultKeyStore(caType);
             logger.atDebug().log("successfully created new CA keystore");
         }
-    }
-
-    /**
-     * Get CA PrivateKey.
-     *
-     * @return                   CA PrivateKey object
-     * @throws KeyStoreException if unable to retrieve PrivateKey object
-     */
-    public PrivateKey getCAPrivateKey() throws KeyStoreException {
+        // Update the certificate chain of the certificate store
+        setCaCertificateChain(keyStore.getCertificateChain(CA_KEY_ALIAS));
+        // Update private key of the certificate store
         try {
-            return (PrivateKey) keyStore.getKey(CA_KEY_ALIAS, getPassphrase());
-        } catch (NoSuchAlgorithmException | UnrecoverableKeyException e) {
+            Key key = keyStore.getKey(CA_KEY_ALIAS, getPassphrase());
+            setCaPrivateKey((PrivateKey) key);
+        } catch (NoSuchAlgorithmException | UnrecoverableKeyException | ClassCastException e) {
             throw new KeyStoreException("unable to retrieve CA private key", e);
         }
     }
 
     /**
      * Get CA Public Certificate.
      *
-     * @return                   CA X509Certificate object
+     * @return CA X509Certificate object
      * @throws KeyStoreException if unable to retrieve the certificate
      */
     public X509Certificate getCACertificate() throws KeyStoreException {
-        return (X509Certificate) keyStore.getCertificate(CA_KEY_ALIAS);
+        return (X509Certificate) getCaCertificateChain()[0];
+    }
+
+    /**
+     * Get CA certificate chain.
+     * This getter method checks if the certificate chain is null before returning it to avoid null pointer exceptions.
+     *
+     * @return caCertificateChain
+     **/
+    @SuppressWarnings("PMD.MethodReturnsInternalArray")
+    public Certificate[] getCaCertificateChain() {
+        if (caCertificateChain == null) {
+            return new Certificate[0];
+        }
+        return caCertificateChain;
+    }
+
+    /**
+     * Get CA private key.
+     * This getter method checks if the private key is null before returning it to avoid null pointer exceptions.
```

Comment:
This statement concerns me. This should never happen as long as the component is in the `RUNNING` state

Reply:
That comment has to be removed.